core/storage/sqlite3_ondisk.rs

```diff
@@ -1564,7 +1565,12 @@ pub fn begin_write_wal_frame(
     };
     #[allow(clippy::arc_with_non_send_sync)]
     let c = Completion::new(CompletionType::Write(WriteCompletion::new(write_complete)));
-    io.pwrite(offset, buffer.clone(), c)?;
+    let res = io.pwrite(offset, buffer.clone(), c);
+    if res.is_err() {
```

Comment:
since we are decrementing the write counter in both success and failure cases, can we instead not use `clone_counter` and just:

```rust
let res = io.pwrite(...);
*write_counter.borrow_mut() -= 1;
res?;
```